plansys2_executor/src/plansys2_executor/ActionExecutor.cpp

```diff
@@ -273,8 +275,15 @@ ActionExecutor::get_params(const std::string & action_expr)
 void
 ActionExecutor::wait_timeout(const ros::WallTimerEvent &event)
 {
-  ROS_WARN("%s -- No action performer for %s. retrying", getName().c_str(), action_.c_str());
-  request_for_performers();
+  if (state_ == RUNNING) {
+    ROS_WARN_STREAM("This timer wasn't properly cancelled");
+    if (waiting_timer_!= nullptr) {
+      waiting_timer_->stop();
+    }
+  } else {
+    ROS_WARN("%s -- No action performer for %s. retrying", getName().c_str(), action_.c_str());
+    request_for_performers();
+  }
```

Comment:
I tried to figure out any logic errors in the PlanSys2 code that could get us into the bad state with `state_ == RUNNING` and the timer not deleted, but I couldn't find any. I'm curious if you actually see the warning message that the timer wasn't canceled.

I'm concerned that there may be some deeper problem we don't yet understand because the original state machine logic seemed ok. Like maybe there is some kind of concurrency here that we're not aware of. (ROS may execute these different callbacks in different threads but all the callbacks are supposed to be guarded by a mutex so they can't run in parallel.)

Anyway, the changes you made seem very low-risk and if they are fixing the theoretically impossible problem in practice this is a good improvement.

Reply:
Yeah I get that warning message all the time to the point the sim is unusable, hence this PR.
I also couldn't find any obvious error in logic, so that's why this PR has some weird redundancies :/